test/test_rpc.py

```diff
@@ -1,64 +1,201 @@
+            date_str=date_str,
+            worker_indent=" " * (WorkerContext.worker_id() + 1) * 2,
+            worker_name=WorkerContext.worker_name(),
+        ),
+        *args,
+        **kwargs
+    )
+
+
+def watch_call(old_func):
+    """Decorate on functions that we would liked to have traces.
+        e.g. `dist.rpc = watch_call(dist.rpc)``
+    """
+
+    @functools.wraps(old_func)
+    def new_func(*args, **kwargs):
+
+        func_name = old_func.__name__
+        sprint(
+            "-> {func_name} Enter ...\n"
+            "  args={args}\n"
+            "  kwargs={kwargs}".format(func_name=func_name, args=args, kwargs=kwargs)
+        )
+        return_value = old_func(*args, **kwargs)
+        sprint("<- {func_name} Exit ...".format(func_name=func_name))
+        return return_value
+
+    return new_func
+
+
+def watch_rpc(old_func):
+    @functools.wraps(old_func)
+    def new_func(to, func, args=None, kwargs=None, async_call=False):
+        sprint(
+            "RPCall {func_name} to {to},\n"
+            "  args={args}\n"
+            "  kwargs={kwargs}".format(
+                func_name=func.__name__, to=to, args=args, kwargs=kwargs
+            )
+        )
+        return old_func(to, func, args=args, kwargs=kwargs, async_call=async_call)
+
+    return new_func
+
+
+dist.rpc = watch_rpc(dist.rpc)
+
+
+def my_function(a, b, c):
+    return a + b + c
+
+
+class my_class:
+    def __init__(self, a):
+        self.a = a
+
+    def my_instance_method(self, b):
+        return self.a + b
+
+    @classmethod
+    def my_class_method(cls, d, e):
+        return d + e
+
+    @staticmethod
+    def my_static_method(f):
+        return f > 10
+
+
+def no_result():
+    print("do nothing")
+
+
+global_var = False
+
+
+@watch_call
+def modify_global_var(global_var_in):
+    global global_var
+    global_var = global_var_in
+
+
+@watch_call
+def nested_rpc(src_name, ttl, reduce_ttl_on):
+    if WorkerContext.worker_name() == reduce_ttl_on:
+        ttl -= 1
+    if ttl > 0:
+        dist.rpc(
+            to=src_name,
+            func=nested_rpc,
+            args=(WorkerContext.worker_name(), ttl, reduce_ttl_on),
+        )
+
+
 # load_tests from common_utils is used to automatically filter tests for
 # sharding on sandcastle. This line silences flake warnings
 load_tests = load_tests
 
-
 if not dist.is_available():
-    print('c10d not available, skipping tests')
+    print("c10d not available, skipping tests")
     sys.exit(0)
 
 
 def _wrap_with_rpc(func):
     def wrapper(self):
         store = dist.FileStore(self.file.name, self.world_size)
-        dist.init_process_group(backend='gloo', rank=self.rank,
-                                world_size=self.world_size, store=store)
-        dist.init_rpc('worker%d' % self.rank)
+        dist.init_process_group(
+            backend="gloo", rank=self.rank, world_size=self.world_size, store=store
+        )
+        WorkerContext(worker_id=self.rank, world_size=self.world_size)
```

Comment:
nit: After this line, `WorkerContext. _INITIALIZED` is already `True`?  Should we actually put this after `init_rpc`?